### PR TITLE
Change logserver Deployment update strategy to Recreate

### PR DESCRIPTION
**Pull request by jiacheliu3**
Addresses https://github.com/Alluxio/alluxio/issues/13422

### CODE REVIEW

**Comment by jiacheliu3** [comment]
FYI @nirav-chotai

File: integration/kubernetes/helm-chart/alluxio/templates/logserver/deployment.yaml [hunk 1]
```diff
@@ -26,6 +26,8 @@ metadata:
     role: alluxio-logserver
 spec:
   replicas: 1
+  strategy:
+    type: Recreate
```
Comment (madanadit) [inline integration/kubernetes/helm-chart/alluxio/templates/logserver/deployment.yaml:30]: Add a comment for this type because of an RWO PV?

**Review by madanadit — APPROVED** [review]
Thx!

**Comment by nirav-chotai** [comment]
@jiacheliu3 , this should not be hardcoded but an option within `values.yaml` with the default being `Recreate`

Note should be added:

- If you're using PVC with RWO access mode, then keep the strategy to `Recreate`. In this case, `logserver` will be down for some time and logs will be lost for that period of time.
- If you're using PVC with RWX access mode, then keep the strategy to `RollingUpdate` In this case, `logserver` will be up and running all the time, so logs will not be lost even while updating `logserver`.

In our DEV environment, I am planning to keep `Recreate` strategy with CephRBD and RWO access mode. However, in the PROD environment, I am going to use `RollingUpdate` strategy with CephFS and RWX access mode.

**Review by ZhuTopher — APPROVED** [review]
LGTM

**Comment by LuQQiu** [comment]
@jiacheliu3 can this PR be merged?

**Comment by jiacheliu3** [comment]
> @jiacheliu3 can this PR be merged?

@LuQQiu As Nirav mentioned, I think it makes sense to make this a configuration instead of hard code. I will do that and request for another round of reviews.

**Comment by jiacheliu3** [comment]
@ZhuTopher @madanadit Please take another look after I extracted the field to a configuration. Thanks!

**Review by ZhuTopher — APPROVED** [review]
LGTM

**Review by madanadit — APPROVED** [review]
Thanks @jiacheliu3

**Comment by jiacheliu3** [comment]
alluxio-bot, merge this please

**Pull request merged**